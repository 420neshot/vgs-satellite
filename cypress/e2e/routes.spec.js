describe('Localhoste route flow', function() {

  beforeEach(() => {
    cy.server();
    cy.route('GET', 'route').as('getRoutes');
    cy.route('PUT', 'route/**').as('putRoute');
    cy.route('DELETE', 'route/**').as('deleteRoute');
    cy.cleanupRoutes();
    cy.fixCypressSpec(__filename);
  });

  it('Visits Localhoste, imports route, then updates and deletes it', function() {
    cy.visit('/routes');
    cy.wait('@getRoutes');
    
    cy.get('[data-role="routes-container"]').contains('There are currently no routes');
    cy.get('[data-role="import-from-yaml"]').attachFile('outbound-redact.yaml');
    cy.wait(['@putRoute', '@getRoutes']);
    cy.url().should('contain', '/edit');
    cy.get('[data-role="edit-route-form"]');

    cy.visit('/routes');
    cy.wait('@getRoutes');

    cy.get('[data-role="routes-type-switch"]').contains('Inbound').click();
    cy.get('[data-role="routes-container"] .tab-pane.active').contains('No Inbound routes');
    
    cy.get('[data-role="routes-type-switch"]').contains('Outbound').click();
    cy.get('[data-role="routes-container"] .tab-pane.active [data-role="route-item"]');

    cy.get('[data-role="routes-type-switch"]').contains('All').click();
    cy.get('[data-role="routes-container"] .tab-pane.active').contains('No Inbound routes');
    cy.get('[data-role="routes-container"] .tab-pane.active [data-role="route-item"]');
    
    cy.get('.route-item .filter-toggler').first().click();

    cy.get('[data-role="route-item"] button').contains('Manage').click({ force: true });
    cy.url().should('contain', '/edit');
    cy.get('[data-role="edit-route-name"]').clear().type('test-basic-redact-renamed');
    cy.get('[data-role="edit-route-save-button"]').click();
    cy.get('.diffsnippet').contains('name: test-basic-redact-renamed');
    cy.get('[data-role="apply-route-changes"]').click();
    cy.wait(['@putRoute', '@getRoutes']);
    cy.get('.iziToast').contains('Your route was updated');

    cy.get('[data-role="edit-route-delete-button"]').click();
    cy.get('[data-role="route-delete-confirm-modal"]').toMatchSnapshot();
    cy.get('[data-role="close-modal"]').click();
    cy.get('button').contains('Cancel').click({ force: true });
    cy.wait('@getRoutes');

    cy.get('[data-role="route-item-name-value"]').contains('test-basic-redact-renamed');
    cy.get('[data-role="route-item"] button').contains('Delete route').click({ force: true });
    cy.contains('delete this route?');
    cy.get('[data-role="route-delete-confirm-modal"]').toMatchSnapshot();

    cy.get('[data-role="modal-submit-button"]').contains('Delete').click();
    cy.wait(['@deleteRoute', '@getRoutes']);

    cy.contains('There are currently no routes');
  });
});
