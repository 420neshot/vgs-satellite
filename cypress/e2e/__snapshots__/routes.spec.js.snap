exports[`Localhoste route flow > Visits Localhoste, imports route, then updates and deletes it #0`] = `
<div
  data-role="route-delete-confirm-modal"
  class="modal-dialog"
  role="document"
>
  <div class="modal-content">
    <div class="modal-header">
      <h5 class="modal-title">Are you sure you want to delete this route?</h5>
      <span id=""><i class="fa-times fas"></i></span>
    </div>
    <div class="modal-body">
      <div>
        <div class="alert alert-warning fade" role="alert">
          <div>
            <p class="m-0"></p>
            <div class="row w-100 mx-0">
              <div class="col-12">
                <div class="row">
                  <div
                    class="text-mono align-self-center"
                    data-role="route-summary-item"
                  >
                    <strong>Redact</strong>&nbsp;the
                    <strong>JSON path</strong>&nbsp;entries
                    <strong>$.foo</strong>&nbsp;in the
                    <strong>request</strong>&nbsp;targets: <strong>body</strong>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <p class="m-0 text-muted">
          This route will be deleted immediately. There is no undo.
        </p>
      </div>
    </div>
    <div class="modal-footer">
      <button
        type="button"
        data-role="modal-submit-button"
        class="btn btn-link"
      >
        Delete</button
      ><button type="button" data-role="close-modal" class="btn btn-primary">
        Cancel
      </button>
    </div>
  </div>
</div>
`;

exports[`Localhoste route flow > Visits Localhoste, imports route, then updates and deletes it #1`] = `
<div
  data-role="route-delete-confirm-modal"
  class="modal-dialog"
  role="document"
>
  <div class="modal-content">
    <div class="modal-header">
      <h5 class="modal-title">Are you sure you want to delete this route?</h5>
      <span id=""><i class="fa-times fas"></i></span>
    </div>
    <div class="modal-body">
      <div>
        <div class="alert alert-warning fade" role="alert">
          <div>
            <p class="m-0"></p>
            <div class="row w-100 mx-0">
              <div class="col-12">
                <div class="row">
                  <div
                    class="text-mono align-self-center"
                    data-role="route-summary-item"
                  >
                    <strong>Redact</strong>&nbsp;the
                    <strong>JSON path</strong>&nbsp;entries
                    <strong>$.foo</strong>&nbsp;in the
                    <strong>request</strong>&nbsp;targets: <strong>body</strong>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <p class="m-0 text-muted">
          This route will be deleted immediately. There is no undo.
        </p>
      </div>
    </div>
    <div class="modal-footer">
      <button
        type="button"
        data-role="modal-submit-button"
        class="btn btn-link"
      >
        Delete</button
      ><button type="button" data-role="close-modal" class="btn btn-primary">
        Cancel
      </button>
    </div>
  </div>
</div>
`;
